src/sig_stfl/sig_stfl.h

```diff
@@ -0,0 +1,659 @@
+ *
+ * @note If the `unlock` function and `mutex` are both set, proceed to unlock the private key.
+ */
+OQS_STATUS OQS_SIG_STFL_SECRET_KEY_unlock(OQS_SIG_STFL_SECRET_KEY *sk);
+
+/**
+ * Set the callback and context for securely storing a stateful secret key.
+ *
+ * This function is designed to be called after a new stateful secret key
+ * has been generated. It enables the library to securely store secret key
+ * and update it every time a Signing operation occurs.
+ * Without properly setting this callback and context, signature generation
+ * will not succeed as the updated state of the secret key cannot be preserved.
+ *
+ * @param[in] sk Pointer to the stateful secret key to be managed.
+ * @param[in] store_cb Callback function that handles the secure storage of the key.
+ * @param[in] context Application-specific context that assists in the storage of secret key data.
+ *                    This context is managed by the application, which allocates it, keeps track of it,
+ *                    and deallocates it as necessary.
+ * @return None.
+ */
+OQS_API void OQS_SIG_STFL_SECRET_KEY_SET_store_cb(OQS_SIG_STFL_SECRET_KEY *sk, secure_store_sk store_cb, void *context);
+
+/**
+ * Serialize the stateful secret key data into a byte array.
+ *
+ * Converts an OQS_SIG_STFL_SECRET_KEY object into a byte array for storage or transmission.
+ *
+ * @param[out] sk_buf_ptr Pointer to the allocated byte array containing the serialized key.
+ * @param[out] sk_len Length of the serialized key byte array.
+ * @param[in] sk Pointer to the OQS_SIG_STFL_SECRET_KEY object to be serialized.
+ * @return OQS_SUCCESS on success, or an OQS error code on failure.
+ *
+ * @note The function allocates memory for the byte array, and it is the caller's responsibility to free this memory after use.
+ */
+OQS_API OQS_STATUS OQS_SECRET_KEY_STFL_serialize_key(uint8_t **sk_buf_ptr, size_t *sk_len, const OQS_SIG_STFL_SECRET_KEY *sk);
+
+/**
+ * Deserialize a byte array into an OQS_SIG_STFL_SECRET_KEY object.
+ *
+ * Transforms a binary representation of a secret key into an OQS_SIG_STFL_SECRET_KEY structure.
+ * After deserialization, the secret key object can be used for subsequent cryptographic operations.
+ *
+ * @param[out] sk A pointer to the secret key object that will be populated from the binary data.
+ * @param[in] key_len The length of the binary secret key data in bytes.
+ * @param[in] sk_buf The buffer containing the serialized secret key data.
+ * @param[in] context Application-specific data used to maintain context about the secret key.
+ * @return OQS_SUCCESS if deserialization was successful; otherwise, OQS_ERROR.
+ *
+ * @attention The caller is responsible for freeing the `sk_buf` memory when it is no longer needed.
+ */
+OQS_API OQS_STATUS OQS_SECRET_KEY_STFL_deserialize_key(OQS_SIG_STFL_SECRET_KEY *sk, size_t key_len, const uint8_t *sk_buf, void *context);
```

Comment:
@SWilson4 one last nit, then I'll approve. What are your thoughts on `key_len` and `sk_buf` getting swapped in order and renamed so they are closer to `serialize`?

```suggestion
OQS_API OQS_STATUS OQS_SECRET_KEY_STFL_deserialize_key(OQS_SIG_STFL_SECRET_KEY *sk, const uint8_t *sk_buf, size_t sk_len, void *context);
```

Reply:
Makes sense to me.

Reply:
Can we rename `key_len` to `sk_len`? `key` is 'undefined' atm. Or even if we rename it to `sk_buf_len` or something. :)

Reply:
But I prefer `sk_len` because that's what it's called in `serialize`

Reply:
Whoops, sorry I missed that half of the comment. I've updated the code to use `sk_buf_len` in all public APIs to make it explicit that the length refers to the buffer and not the secret key object.